src/jsCore.ts

```diff
@@ -0,0 +1,497 @@
+  }
+}
+
+function trackSelfDescribingEvent(details: {
+  tracker: string | null;
+  eventData: SelfDescribing;
+  contexts: EventContext[];
+}): Promise<void> {
+  forTracker(details.tracker, (tracker) => {
+    tracker.track(
+      buildSelfDescribingEvent({
+        event: details.eventData,
+      }),
+      details.contexts
+    );
+  });
+  return <Promise<void>>Promise.resolve();
+}
+
+function trackStructuredEvent(details: {
+  tracker: string | null;
+  eventData: StructuredProps;
+  contexts: EventContext[];
+}): Promise<void> {
+  forTracker(details.tracker, (tracker) => {
+    tracker.track(buildStructEvent(details.eventData), details.contexts);
+  });
+  return <Promise<void>>Promise.resolve();
+}
+
+function trackScreenViewEvent(details: {
+  tracker: string | null;
+  eventData: ScreenViewProps;
+  contexts: EventContext[];
+}): Promise<void> {
+  forTracker(details.tracker, (tracker) => {
+    tracker.track(buildScreenView(details.eventData), details.contexts);
+  });
+  return <Promise<void>>Promise.resolve();
+}
+
+function trackPageViewEvent(details: {
+  tracker: string | null;
+  eventData: PageViewProps;
+  contexts: EventContext[];
+}): Promise<void> {
+  forTracker(details.tracker, (tracker) => {
+    tracker.track(buildPageView(details.eventData), details.contexts);
+  });
+  return <Promise<void>>Promise.resolve();
+}
+
+function trackTimingEvent(details: {
+  tracker: string | null;
+  eventData: TimingProps;
+  contexts: EventContext[];
+}): Promise<void> {
+  return trackSelfDescribingEvent({
+    tracker: details.tracker,
+    eventData: {
+      schema: schemas.timingSchema,
+      data: details.eventData,
+    },
+    contexts: details.contexts,
+  });
+}
+
+function trackConsentGrantedEvent(details: {
+  tracker: string | null;
+  eventData: ConsentGrantedProps;
+  contexts: EventContext[];
+}): Promise<void> {
+  forTracker(details.tracker, (tracker) => {
+    const built = buildConsentGranted({
+      id: details.eventData.documentId,
+      version: details.eventData.version,
+      name: details.eventData.name,
+      description: details.eventData.documentDescription,
+      expiry: details.eventData.expiry,
+    });
+    tracker.track(built.event, details.contexts.concat(built.context));
+  });
+  return <Promise<void>>Promise.resolve();
+}
+
+function trackConsentWithdrawnEvent(details: {
+  tracker: string | null;
+  eventData: ConsentWithdrawnProps;
+  contexts: EventContext[];
+}): Promise<void> {
+  forTracker(details.tracker, (tracker) => {
+    const built = buildConsentWithdrawn({
+      all: details.eventData.all,
+      id: details.eventData.documentId,
+      version: details.eventData.version,
+      name: details.eventData.name,
+      description: details.eventData.documentDescription,
+    });
+    tracker.track(built.event, details.contexts.concat(built.context));
+  });
+  return <Promise<void>>Promise.resolve();
+}
+
+function trackEcommerceTransactionEvent(details: {
```

Comment:
Not sure about this, but are the mobile trackers tracking `transaction_item` events automatically besides the `transaction` event? Not sure if we'd like parity here. What do you think?

Reply:
You're right, the mobile trackers track the items as separate events. Change it here as well. Great spot, thanks!